src/main/java/de/retest/recheck/persistence/GitUtil.java

```diff
@@ -0,0 +1,67 @@
+package de.retest.recheck.persistence;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class GitUtil {
+
+	protected static final String GIT_COMMAND_VERSION = "git --version";
+	protected static final String GIT_COMMAND_BRANCH_NAME = "git rev-parse --abbrev-ref HEAD";
+	protected static final String GIT_COMMAND_COMMIT_HASH = "git rev-parse HEAD";
+
+	private final boolean localGit;
+
+	public GitUtil() {
+		localGit = isGitInstalled();
+	}
+
+	protected boolean isGitInstalled() {
+		try {
+			return executeGitCommand( GIT_COMMAND_VERSION ).startsWith( "git" );
+		} catch ( final Exception e ) {
+			log.debug( "`{}` resulted in exception {}, assuming no git installed.", GIT_COMMAND_VERSION,
+					e.getMessage() );
+			return false;
+		}
+	}
+
+	protected String executeGitCommand( final String command ) throws IOException, InterruptedException {
```

Comment:
Just stumbled over this: rather than doing all of this manually, how about using JGit like we do in the rebazer?

Add this dependency:

```xml
<dependency>
	<groupId>org.eclipse.jgit</groupId>
	<artifactId>org.eclipse.jgit</artifactId>
	<version>5.5.1.201910021850-r</version>
</dependency>
```

This snippet should retrieve the name:

```java
final Repository repo = FileRepositoryBuilder.create( new File( "./.git/" ) );
try ( final Git git = new Git( repo ) ) {
	final ObjectId head = repo.resolve( "HEAD" );
	final Map<ObjectId, String> names = git.nameRev() //
			.add( head ) //
			.call();
	final String currentBranchName = names.get( head );
}
```

This should correspond to:

```
$ git name-rev HEAD
```

Maybe this can be done even more elegant?

https://stackoverflow.com/q/6245570

Reply:
I don't like adding yet another dependency to replace a mere 4 lines of code.

Reply:
As RET-1907 shows, the fewer dependencies, the better.

Reply:
> I don't like adding yet another dependency to replace a mere 4 lines of code.

It replaces the content of the `GitUtil`, not just these 4 lines.

> As RET-1907 shows, the fewer dependencies, the better.

You definitely can't say that per se.

Reply:
Hmm. I think "the fewer dependencies, the better" is something that generally holds. It's a always a tradeoff, so you usually get something in return. Just in this case, I think what you get is not worth the tradeoff.

Reply:
It's just a suggestion.

Also note that JGit is a pure Java implemention of Git. Suppose you have a dev that has no Git installed or not in his `PATH`—one can, for example, solely work with [EGit](https://www.eclipse.org/egit/)—JGit still works. In contrast, the current implementation doesn't as far as I can see.

Reply:
> Suppose you have a dev that has no Git installed or not in his PATH

That is a good point. This would then not just solve the manual problem (e.g. addressing this [comment](https://github.com/retest/recheck/pull/551#discussion_r353696258)), but make the execution independent of the installed git.

Reply:
Hmm. I do not want to add 2.6 MB to replace a couple of lines of code. Currently I don't think we need a full-fledged git implementation. KISS.

Reply:
I think the probability that a user encounters the aformentioned issues (no Git or not in `PATH`) is higher than the probability that a user complains about the (fat) JAR size or the number of downloaded dependencies. (Did you ever do that?)

And regarding KISS: I think this is actually a counter-example. To keep it simple, use something that "simply" works out-of-the-box. Why bother with a custom implementation, a local Git installation, OS differences, etc.

However, to end the discussion: your PR works, it is not a bad hack or something. I just think that using JGit has more pros than cons compared to the current implementation. So, it is your decision … but mark my words. 😉

Reply:
In my opinion, it's not about the number of downloaded dependencies, but the potential problems that come with this (again see RET-1907) or the problems we have with one of our clients I mentioned... I'll mark your words, and I am open to replace that ones we really need that or the current solution causes problems. But until then, I'd like to keep it small...